Tests/SwiftGenKitTests/StringsTests.swift

```diff
@@ -58,15 +58,109 @@ class StringsTests: XCTestCase {
+    let errorMessage = try XCTUnwrap(receivedWarning)
+    let expectedMessage = #"""
+    Table "Localizable" already loaded by other parser, the parser for [.strings] files cannot modify existing keys:
+    ObjectOwnership
+    alert__message
+    alert__title
+    apples.count
+    bananas.owner
+    percent
+    private
+    settings.navigation-bar.self
+    settings.navigation-bar.title.deeper.than.we.can.handle.no.really.this.is.deep
+    settings.navigation-bar.title.even.deeper
+    settings.user__profile_section.HEADER_TITLE
+    settings.user__profile_section.footer_text
+    types
+    """#
+
+    XCTAssertEqual(errorMessage, expectedMessage)
+  }
+
+  func testPlurals() throws {
+    let parser = try Strings.Parser()
+    try parser.searchAndParse(path: Fixtures.path(for: "Localizable.stringsdict", sub: .strings))
+
+    let result = parser.stencilContext()
+    XCTDiffContexts(result, expected: "plurals", sub: .strings)
+  }
+
+  func testAdvancedPlurals() throws {
+    let parser = try Strings.Parser()
+    try parser.searchAndParse(path: Fixtures.path(for: "LocPluralAdvanced.stringsdict", sub: .strings))
+
+    let result = parser.stencilContext()
+    XCTDiffContexts(result, expected: "plurals-advanced", sub: .strings)
+  }
+
+// Commented out since nested format strings in plurals are not supported yet.
+//
+//  func testNestedPlurals() throws {
+//    let parser = try Strings.Parser()
+//    try parser.searchAndParse(path: Fixtures.path(for: "LocPluralNested.stringsdict", sub: .strings))
+//
+//    let result = parser.stencilContext()
+//    XCTDiffContexts(result, expected: "plurals-nested", sub: .strings)
+//  }
+
+  func testSameTableWithPluralsParsingStringsFirst() throws {
+    let parser = try Strings.Parser()
+    try parser.searchAndParse(path: Fixtures.path(for: "Localizable.strings", sub: .strings))
+    try parser.searchAndParse(path: Fixtures.path(for: "Localizable.stringsdict", sub: .strings))
+
+    let result = parser.stencilContext()
+    XCTDiffContexts(result, expected: "plurals-same-table-strings-first", sub: .strings)
+  }
+
+  func testSameTableWithPluralsParsingPluralsFirst() throws {
+    let parser = try Strings.Parser()
+    var receivedWarning: String?
+    parser.warningHandler = { message, _, _ -> Void in
+      receivedWarning = message
+    }
+    try parser.searchAndParse(path: Fixtures.path(for: "Localizable.stringsdict", sub: .strings))
+    try parser.searchAndParse(path: Fixtures.path(for: "Localizable.strings", sub: .strings))
+
+    let errorMessage = try XCTUnwrap(receivedWarning)
+    let expectedMessage = #"""
+    Table "Localizable" already loaded by other parser, the parser for [.strings] files cannot modify existing keys:
+    apples.count
+    """#
+
+    XCTAssertEqual(errorMessage, expectedMessage)
+
+    let result = parser.stencilContext()
+    XCTDiffContexts(result, expected: "plurals-same-table-plurals-first", sub: .strings)
```

Comment:
As said in comments, given that there's no guarantee that the user doesn't explicitly list their `.stringsdict` first and their `.strings` next in their config file – or that there's (I think) no guaranteed that the order of parsing the files in a folder is deterministic in our code – in the end we should end up with the exact same output for both `testSameTableWithPluralsParsingStringsFirst` and `testSameTableWithPluralsParsingPluralsFirst`

So maybe you could just use the same "expected" fixture for both those tests (which would be easier to maintain rather than having 2 identical files, and would make it clearer that we expect the same result in both cases)